src/python/pants/engine/rules.py

```diff
@@ -348,8 +347,8 @@ def get_some_union_type(x):
 
```

Comment:
This started erroring out here when running locally and in CI without these (necessary) changes. I'm not sure why this is the case, but that is why this file is changed in this PR.